test/rootfs.jl

```diff
@@ -1,163 +1,145 @@
-    @test supported_microarchitectures(UnknownPlatform()) == []
+    @test expand_cxxstring_abis([Platform("i686", "linux"; cxxstring_abi="cxx11")]) ==
+        [Platform("i686", "linux"; cxxstring_abi="cxx11")]
 
     # expand_microarchitectures
     @test expand_microarchitectures([AnyPlatform()]) == [AnyPlatform()]
-    @test expand_microarchitectures(ExtendedPlatform(Linux(:x86_64); cuda="10.1")) == [
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx", cuda="10.1"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx2", cuda="10.1"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx512", cuda="10.1"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="x86_64", cuda="10.1"),
+    @test sort(expand_microarchitectures(Platform("x86_64", "linux"; cuda="10.1")), by=triplet) == [
+        Platform("x86_64", "linux"; libc="glibc", march="avx", cuda="10.1"),
+        Platform("x86_64", "linux"; libc="glibc", march="avx2", cuda="10.1"),
+        Platform("x86_64", "linux"; libc="glibc", march="avx512", cuda="10.1"),
+        Platform("x86_64", "linux"; libc="glibc", march="x86_64", cuda="10.1"),
     ]
-    @test expand_microarchitectures(filter!(p -> p isa Linux && libc(p) == :glibc, supported_platforms())) == [
-        Linux(:i686, libc=:glibc),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx2"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="avx512"),
-        ExtendedPlatform(Linux(:x86_64, libc=:glibc); march="x86_64"),
-        ExtendedPlatform(Linux(:aarch64, libc=:glibc); march="armv8"),
-        ExtendedPlatform(Linux(:aarch64, libc=:glibc); march="carmel"),
-        ExtendedPlatform(Linux(:aarch64, libc=:glibc); march="thunderx2"),
-        ExtendedPlatform(Linux(:armv7l, libc=:glibc, call_abi=:eabihf); march="armv7l"),
-        ExtendedPlatform(Linux(:armv7l, libc=:glibc, call_abi=:eabihf); march="neon"),
-        ExtendedPlatform(Linux(:armv7l, libc=:glibc, call_abi=:eabihf); march="vfp4"),
-        Linux(:powerpc64le, libc=:glibc),
+    @test sort(expand_microarchitectures(filter!(p -> Sys.islinux(p) && libc(p) == "glibc", supported_platforms())), by=triplet) == [
+        Platform("aarch64", "linux"; libc="glibc", march="armv8_0"),
+        Platform("aarch64", "linux"; libc="glibc", march="armv8_1"),
+        Platform("aarch64", "linux"; libc="glibc", march="armv8_2_crypto"),
+        Platform("aarch64", "linux"; libc="glibc", march="armv8_4_crypto_sve"),
+        Platform("armv7l", "linux"; libc="glibc", call_abi="eabihf", march="armv7l"),
+        Platform("armv7l", "linux"; libc="glibc", call_abi="eabihf", march="neonvfpv4"),
+        Platform("i686", "linux"; libc="glibc", march="prescott"),
+        Platform("powerpc64le", "linux"; libc="glibc", march="power8"),
+        Platform("powerpc64le", "linux"; libc="glibc", march="power9"),
```

Comment:
```suggestion
        Platform("i686", "linux"; libc="glibc", march="i686"),
        Platform("i686", "linux"; libc="glibc", march="prescott"),
        Platform("powerpc64le", "linux"; libc="glibc", march="power8"), 
```
This should make the test pass, but we're missing power9 in the expansion?